sdks/python/apache_beam/examples/avro_nyc_trips_it_test.py

```diff
@@ -0,0 +1,213 @@
+  }
+
+  RECORDS = [
+      {
+          'hvfhs_license_num': 'HV0003',
+          'request_datetime': 1549008086000,
+          'trip_miles': 2.45,
+          'trip_time': 579,
+          'base_passenger_fare': 9.35,
+          'tolls': 0.0,
+          'bcf': 0.23,
+          'sales_tax': 0.83,
+          'congestion_surcharge': 0.0,
+          'airport_fee': None,
+          'tips': 0.0,
+          'driver_pay': 7.48
+      },
+      {
+          'hvfhs_license_num': 'HV0003',
+          'request_datetime': 1549009568000,
+          'trip_miles': 1.71,
+          'trip_time': 490,
+          'base_passenger_fare': 7.91,
+          'tolls': 0.0,
+          'bcf': 0.2,
+          'sales_tax': 0.7,
+          'congestion_surcharge': 0.0,
+          'airport_fee': None,
+          'tips': 2.0,
+          'driver_pay': 7.93
+      },
+      {
+          'hvfhs_license_num': 'HV0005',
+          'request_datetime': 1549010613000,
+          'trip_miles': 11.24,
+          'trip_time': 1739,
+          'base_passenger_fare': 29.77,
+          'tolls': 0.72,
+          'bcf': 0.76,
+          'sales_tax': 2.71,
+          'congestion_surcharge': 0.0,
+          'airport_fee': None,
+          'tips': 0.0,
+          'driver_pay': 22.09
+      },
+      {
+          'hvfhs_license_num': 'HV0005',
+          'request_datetime': 1549010420000,
+          'trip_miles': 5.71,
+          'trip_time': 1559,
+          'base_passenger_fare': 21.69,
+          'tolls': 0.24,
+          'bcf': 0.55,
+          'sales_tax': 1.95,
+          'congestion_surcharge': 0.0,
+          'airport_fee': None,
+          'tips': 0.0,
+          'driver_pay': 14.87
+      },
+  ]
+
+  EXPECTED = [
+      {
+          'service': 'Lyft',
+          'day': 'Fri',
+          'total_price': 58.39,
+          'total_driver_pay': 36.96,
+          'total_trip_miles': 16.95,
+          'total_trip_minutes': 54.96666666666667,
+          'total_number_of_trips': 2,
+          'price_per_trip': 29.195,
+          'price_per_mile': 3.4448377581120946,
+          'price_per_minute': 1.0622801697998787,
+          'driver_pay_per_trip': 18.48,
+          'driver_pay_per_mile': 2.1805309734513276,
+          'driver_pay_per_minute': 0.6724075197089144,
+          'miles_per_hour': 18.502122498483928
+      },
+      {
+          'service': 'Uber',
+          'day': 'Fri',
+          'total_price': 21.22,
+          'total_driver_pay': 17.41,
+          'total_trip_miles': 4.16,
+          'total_trip_minutes': 17.816666666666666,
+          'total_number_of_trips': 2,
+          'price_per_trip': 10.61,
+          'price_per_mile': 5.100961538461538,
+          'price_per_minute': 1.1910196445275958,
+          'driver_pay_per_trip': 8.705,
+          'driver_pay_per_mile': 4.185096153846153,
+          'driver_pay_per_minute': 0.9771749298409729,
+          'miles_per_hour': 14.00935453695042
+      },
+  ]
+
+  # TODO Enable when fixed this tests for Dataflow runner
```

Comment:
I was looking at this test case to write my test case:
https://github.com/apache/beam/blob/96989dc9b11457e5d03bb145877faa5f0869db23/sdks/python/apache_beam/examples/complete/top_wikipedia_sessions_it_test.py#L72-L75

Although, I changed the input/output file name to start with `/tmp` rather than `gs://` which I think makes it use the `TestDirectRunner` rather than the `TestDataflowRunner` to my understanding. So, maybe I can remove this line and the @pytest.mark.sickbay_dataflow decorator`?

Reply:
both Direct and Dataflow runners should be able to read the public GCS bucket, so you could use the input from example instead of regenerating.  

Instead of hardcoding the output location in the `..._it_test.py`, you can query it from TestPipeline, like done in https://github.com/apache/beam/blob/e1396db4b18ecd60c76ca3dc271238335218c81c/sdks/python/apache_beam/examples/wordcount_it_test.py#L121 

The option named 'output' is being set in:
https://github.com/apache/beam/blob/e1396db4b18ecd60c76ca3dc271238335218c81c/sdks/python/scripts/run_integration_test.sh#L235 , I think this is script is used both in Direct and Dataflow runners.

Reply:
> There is a lot of other things in that wordcount integration test that I'm not sure if I need to copy as well or not? Like the mocking of the Google Cloud credentials.

It's ok, you don't need that.